### DISCUSSION TITLE
best way to get token without localstorage

### DISCUSSION THREAD

**Discussion by oottoohh** [infinitered/ignite#2324] [Q&A] [answered]
is it best way, every we need hit API we always get from localstorage ? in my opinion maybe something problem with performance if we do this cause we always communicate with localstorage every consume API, i hope im wrong.. so if this is not the best way can you give me solution how to get token auth using mobx without communicate with localstorage ?

**Comment by ryskin** [comment, marked as answer]
if you use persistence in your store, you write data to store for every change and that's ok,

**Comment by ryskin** [comment]
if you worry about the performance of AsyncStorage you can use MMKV ~30x faster

**Reply by ryskin** [reply]
Also, you need to store token only in case you close the app, this will reset your store, and you can store your token in mobx store and make the store persistent but it should be anyway stored somewhere permanently to restore if you close app.
